maths/gcd_of_n_numbers.py

```diff
@@ -0,0 +1,76 @@
+"""
+Gcd of N Numbers
+Reference: https://en.wikipedia.org/wiki/Greatest_common_divisor
+"""
+
+from collections import Counter
+
+
+def get_factors(
+    number: int, factors: Counter | None = None, factor: int = 2
+) -> Counter:
+    """
+    this is a recursive function for get all factors of number
+    >>> get_factors(45)
+    Counter({3: 2, 5: 1})
+    >>> get_factors(2520)
+    Counter({2: 3, 3: 2, 5: 1, 7: 1})
+    >>> get_factors(23)
+    Counter({23: 1})
+
+    factor can be all numbers from 2 to number that we check if number % factor == 0
+    if it is equal to zero, we check again with number // factor
+    else we increase factor by one
+    """
+
+    if not isinstance(number, int):
+        raise TypeError("number must be integer")
+
+    if factors is None:
+        # first call of get factors and so there are no founded factors
+        factors = Counter()
```

Comment:
All falsey values are as dangerous as None.
```suggestion
    factors = factors or  Counter()
```